gradio/external.py

```diff
@@ -59,6 +60,13 @@ def load_blocks_from_repo(
         factory_methods.keys()
     )
 
```

Comment:
In serializing.py, `FileSerializable`, we should be using the `proxy` route, no?

Line 167:
```python
        elif isinstance(x, dict):
            if x.get("is_file", False):
                if root_url is not None:
                    x["name"] = urllib.parse.urljoin(root_url, "file=" + x["name"])
```

Reply:
Hmmm yes I think that might fix the issue of using as a function too. Nice!

Reply:
Hmm actually this is a bit tricky since the fastapi server is not necessarily launched at this point. So there's not necessarily a server available that you can use to proxy

Reply:
Maybe the problem is that when we go to deserialize the file, `validate_url` returns False since the authorization headers are not set?

```python
def validate_url(possible_url: str) -> bool:
    headers = {"User-Agent": "gradio (https://gradio.app/; team@gradio.app)"}
    try:
        head_request = requests.head(possible_url, headers=headers)
        if head_request.status_code == 405:
            return requests.get(possible_url, headers=headers).ok
        return head_request.ok
    except Exception:
        return False
```

I'm testing with this space: https://huggingface.co/spaces/gradio-tests/not-actually-private-space-output-audio and if I follow the stacktrace I think it's due to `validate_url` being False even though the URL is valid https://gradio-tests-not-actually-private-space-outpu-3b22378.hf.space/file=/tmp/tmp3bgq_vf3.wav

Reply:
You're right, that's part of the problem. We basically need to download the file with the authorization headers and then save it locally so that it can be accessed by an end user without needing to make another request. I know what needs to be done, just a matter of implementing it correctly.